File: flightsBackend/src/main/java/com/example/flightsBackend/entities/flightData/Segments.java
package com.example.flightsBackend.entities.flightData;

import lombok.Data;

@Data
public class Segments {

    private Departure departure;
    private Arrival arrival;
    private String carrierCode;
    private String number;
    private Aircraft aircraft;
    private Operating operating;
    private String duration;
    private String id;
    private Integer numberOfStops;
    private Boolean blacklistedInEU;

}

File: flightsFrontend/src/app/model/flightData/segments.ts
import { Aircraft } from './aircraft';
import { Arrival } from './arrival';
import { Departure } from './departure';
import { Operating } from './operating';

export class Segments {
    departure: Departure;
    arrival: Arrival;
    carrierCode: string;
    numb: string;
    aircraft: Aircraft;
    operating: Operating;
    duration: string;
    id: string;
    numberOfStops: number;
    blacklistedInEU: boolean;
}
